patches/chromium/feat_add_streaming-protocol_registry_to_multibuffer_data_source.patch

```diff
@@ -0,0 +1,87 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Paul Frazee <pfrazee@gmail.com>
+Date: Mon, 11 May 2020 16:44:50 -0500
+Subject: feat: add streaming-protocol registry to multibuffer_data_source
+
+blink::WebMediaPlayerImpl - which provides the <video> and <audio> behaviors - needs to know
+whether a data source will stream or fully buffer the response. It determines this behavior
+with MultibufferDataSource::AssumeFullyBuffered() which has http/s hardwired. An incorrect
+determination will cause the video/audio to fail playing.
+
+This patch adds a list of "streaming protocols" to the MultibufferDataSource in order to allow
+other protocols to register their streaming behavior. MultibufferDataSource::AssumeFullyBuffered()
+then refers to the list so that it can correctly determine the data source's settings.
+
+diff --git a/media/blink/multibuffer_data_source.cc b/media/blink/multibuffer_data_source.cc
+index 0f6ae1fb8b4ff9f24ce3f407b7359e016fc6de5f..579e8448a1319e7953abb36f0e601debfbc6602c 100644
+--- a/media/blink/multibuffer_data_source.cc
++++ b/media/blink/multibuffer_data_source.cc
+@@ -10,6 +10,7 @@
+ #include "base/callback_helpers.h"
+ #include "base/location.h"
+ #include "base/macros.h"
++#include "base/no_destructor.h"
+ #include "base/numerics/ranges.h"
+ #include "base/numerics/safe_conversions.h"
+ #include "base/single_thread_task_runner.h"
+@@ -65,10 +66,31 @@ const int kUpdateBufferSizeFrequency = 32;
+ // How long to we delay a seek after a read?
+ constexpr base::TimeDelta kSeekDelay = base::TimeDelta::FromMilliseconds(20);
+ 
++// used to run one-time setup on the streaming schemes list
++bool g_streaming_schemes_initialized = false;
++
++std::vector<std::string>* GetStreamingSchemes() {
++  static base::NoDestructor<std::vector<std::string>> streaming_schemes;
++
++  // initialize default schemes
++  std::vector<std::string>* schemes = streaming_schemes.get();
++  if (!g_streaming_schemes_initialized) {
++    schemes->push_back(url::kHttpsScheme);
++    schemes->push_back(url::kHttpScheme);
++    g_streaming_schemes_initialized = true;
++  }
```

Comment:
I think this can be done with an initializer, e.g.
```suggestion
+  static base::NoDestructor<std::vector<std::string>> streaming_schemes = {url::kHttpsScheme, url::kHttpScheme};
```

Reply:
Aha, yes you can if you pass the initialization value into the constructor. Just pushed an update.